### PR TITLE
node: split out initializers into setup.go

### PR DESCRIPTION
**Pull request by cmwaters**
node.go is a big messy file. It's also in many ways the entry point for anyone wanting to learn the codebase. This is the first in a few PR's in trying to clean it up. To increase legibility I have split out all the initializer functions and put them in a separate setup.go file. I've also restructured the file so that the node instance is the first thing you see, followed by the constructor/s and the `OnStart`, `OnStop` methods

### CODE REVIEW

**Comment by cmwaters** [comment]
> Would really like to see the new repo structure first prior to doing an intense cleanup of the node

Yeah this makes sense to me too

File: node/setup.go [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+
+	privatePeerIDs := make(map[p2p.NodeID]struct{})
+	for _, id := range strings.SplitAndTrimEmpty(config.P2P.PrivatePeerIDs, ",", " ") {
+		privatePeerIDs[p2p.NodeID(id)] = struct{}{}
+	}
+
+	options := p2p.PeerManagerOptions{
+		MaxConnected:           maxConns,
+		MaxConnectedUpgrade:    4,
+		MaxPeers:               1000,
```
[the remark this answers is not in the document]
Comment (creachadair) [reply node/setup.go:428]: That appears to have been added in https://github.com/tendermint/tendermint/pull/6056. From the comments there, it sounds like this was meant to become configurable, but perhaps that never happened.

[the remark this answers is not in the document]
Comment (creachadair) [reply node/setup.go:428]: (Note that in this PR, the line already existed—it's just being moved from one file to another)

[the remark this answers is not in the document]
Comment (evan-forbes) [reply node/setup.go:428]: That makes sense and what I figured happened. 

🙏 I'll open an issue and PR

**Pull request merged**